plugins/autofarm.cpp

```diff
@@ -213,11 +203,11 @@ class AutoFarm {
             farm->plant_id[season] = new_plant_id;
             out << "autofarm: changing farm #" << farm->id <<
                 " from " << get_plant_name(old_plant_id) <<
-                " to " << get_plant_name(new_plant_id) << endl;
+                " to " << get_plant_name(new_plant_id) << '\n';
         }
     }
 
-    void set_farms(color_ostream& out, set<int> plants, vector<df::building_farmplotst*> farms)
+    void set_farms(color_ostream& out, std::set<int>& plants, std::vector<df::building_farmplotst*>& farms)
```

Comment:
plants could be made const. I think farms can too. That way the change to passing by reference can guarantee no side effects.